packages/eas-cli/src/submit/android/ServiceAccountSource.ts

```diff
@@ -30,66 +31,115 @@ interface ServiceAccountPromptSource extends ServiceAccountSourceBase {
   sourceType: ServiceAccountSourceType.prompt;
 }
 
-interface ServiceAccountDetectSource extends ServiceAccountSourceBase {
-  sourceType: ServiceAccountSourceType.detect;
+export interface ServiceAccountCredentialsServiceSource extends ServiceAccountSourceBase {
+  sourceType: ServiceAccountSourceType.credentialsService;
 }
 
+export type ServiceAccountKeyResult = {
+  result: ServiceAccountKeyFile | ServiceAccountKeyFromExpoServers;
+  summary: ServiceAccountKeySummary;
+};
+
+type ServiceAccountKeySummary = {
+  source: 'local' | 'EAS servers';
+  path?: string;
+  email: string;
+};
+
+type ServiceAccountKeyFile = {
+  googleServiceAccountKeyJson: string;
+};
+
+type ServiceAccountKeyFromExpoServers = {
+  googleServiceAccountKeyId: string;
+};
+
 export type ServiceAccountSource =
   | ServiceAccountPathSource
   | ServiceAccountPromptSource
-  | ServiceAccountDetectSource;
+  | ServiceAccountCredentialsServiceSource;
+
+export async function getServiceAccountKeyResultAsync(
+  ctx: SubmissionContext<Platform.ANDROID>,
+  source: ServiceAccountSource,
+  androidApplicationIdentifier: string
+): Promise<ServiceAccountKeyResult> {
+  if (source.sourceType === ServiceAccountSourceType.credentialsService) {
+    return await getServiceAccountFromCredentialsServiceAsync(ctx, androidApplicationIdentifier);
+  }
+  return await getServiceAccountLocallyAsync(source);
```

Comment:
```suggestion
  } else {
    return await getServiceAccountLocallyAsync(source);
  }
```
This is my personal preference but I really don't like using "early return" for things like that. 
I think it's fine when using for throwing errors. However, in this case, the part of the code inside the `if` (`getServiceAccountFromCredentialsServiceAsync` call) is not a special path.